edgelet/edgelet-docker/src/notary.rs

```diff
@@ -0,0 +1,185 @@
+        ))
+    })?;
+    debug!("Config JSON file created successfully");
+    Ok(config_file_path)
+}
+
+pub fn notary_lookup(
+    notary_auth: &str,
+    image_gun: &str,
+    tag: &str,
+    config_path: &str,
+    lock: tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+) -> impl Future<
+    Item = (
+        String,
+        tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+    ),
+    Error = Error,
+> {
+    Command::new("notary")
+        .env("NOTARY_AUTH", notary_auth)
+        .arg("lookup")
+        .args(&[image_gun, tag])
+        .args(&["-c", config_path])
+        .output_async()
+        .then(|notary_output| {
+            let notary_output = notary_output.with_context(|_| {
+                ErrorKind::LaunchNotary("could not spawn notary process".to_owned())
+            })?;
+            let notary_output_string =
+                String::from_utf8(notary_output.stdout).with_context(|_| {
+                    ErrorKind::LaunchNotary("could not retrieve notary output as string".to_owned())
+                })?;
+            let split_array: Vec<&str> = notary_output_string.split_whitespace().collect();
+            if split_array.len() < 2 {
+                return Err(ErrorKind::LaunchNotary(
+                    "notary digest split array is empty".to_owned(),
+                )
+                .into());
+            }
+
+            // Notary Server output on lookup is of the format [tag, digest, bytes]
+            Ok((split_array[1].to_string(), lock))
+        })
+}
+
+#[cfg(test)]
+mod test {
+    #[cfg(target_os = "linux")]
+    use crate::notary;
+    #[cfg(target_os = "linux")]
+    use crate::ErrorKind;
+    #[cfg(target_os = "linux")]
+    use std::path::PathBuf;
+    #[cfg(target_os = "linux")]
+    use tempfile::NamedTempFile;
+
+    #[cfg(unix)]
+    #[test]
+    fn check_for_empty_hostname() {
+        let registry_server_hostname = String::new();
+        let home_dir = NamedTempFile::new().unwrap();
+        let root_ca_file = NamedTempFile::new().unwrap();
+        let result = notary::notary_init(
+            home_dir.path(),
+            &registry_server_hostname,
+            &root_ca_file.path().to_path_buf(),
+        );
+        let err = result.unwrap_err();
+        assert!(matches!(err.kind(), ErrorKind::InitializeNotary(s) if s == "hostname is empty"));
+    }
+
+    #[cfg(unix)]
+    #[test]
+    fn check_for_root_ca_file_does_not_exist() {
+        let registry_server_hostname = r"myregistry.azurecr.io";
+        let home_dir = NamedTempFile::new().unwrap();
+        let root_ca_file = PathBuf::from("filedoesnotexist.crt");
+        let result = notary::notary_init(home_dir.path(), &registry_server_hostname, &root_ca_file);
+        let err = result.unwrap_err();
+        let _display_msg = format!("root ca at path {} does not exist", root_ca_file.display());
+        assert!(matches!(
+            err.kind(),
+            ErrorKind::InitializeNotary(_display_msg)
```

Comment:
This is not doing what you think it's doing.

This is equivalent to writing

```rust
assert!(match err.kind() {
    ErrorKind::InitializeNotary(_display_msg) => true,
    _ => false,
});
```

which means `_display_msg` is just a pattern that is being ignored.

You want `ErrorKind::InitializeNotary(m) where m == display_msg` to actually assert that the string in the enum variant is `display_msg`